podvm/files/etc/systemd/system/kata-agent.service

```diff
@@ -5,9 +5,11 @@ Wants=process-user-data.service
 After=netns@podns.service process-user-data.service
 
 [Service]
-ExecStart=/usr/local/bin/kata-agent --config /etc/agent-config.toml
+# Temporary fix for
+ExecStartPre=mkdir -p /run/kata-containers
+ExecStart=/usr/local/bin/kata-agent --config /run/peerpod/agent-config.toml
```

Comment:
A question came to my mind when going through the changes again.
 
@katexochen can the systemd config files for mkosi based builds be kept under podvm-mkosi directory? 

This will give us sometime to cleanly deprecate packer based builds and eventually remove packer based builds or move it out to separate repo.

Also having the required dependencies for mkosi under separate directory will make subsequent changes to mkosi based builds easier.

Reply:
> katexochen can the systemd config files for mkosi based builds be kept under podvm-mkosi directory?

Not sure I understand your proposal here. You don't want this PR to have any changes inside of `/podvm` ? The binaries container files for mkosi are in that directory, and mkosi uses the same `/podvm/files` as the legacy images.

There is no way scoping these changes to `/podvm-mkosi`, as other components are touched, like process-user-data, adaptor and forwarder, as well as the guest-components.

Reply:
> > katexochen can the systemd config files for mkosi based builds be kept under podvm-mkosi directory?
> 
> Not sure I understand your proposal here. You don't want this PR to have any changes inside of `/podvm` ? The binaries container files for mkosi are in that directory, and mkosi uses the same `/podvm/files` as the legacy images.
> 

Yes, I'm looking at having all mkosi related configs and deps under podvm-mkosi to help with deprecating packer based builds. 
I created a related issue to track this - https://github.com/confidential-containers/cloud-api-adaptor/issues/1615


> There is no way scoping these changes to `/podvm-mkosi`, as other components are touched, like process-user-data, adaptor and forwarder, as well as the guest-components.

I'm only referring to configs (like the systemd scripts, Dockerfile, Makefile). Apologise if it was not clear.

At a high level, this is what I was thinking
- Copy the systemd config files in `podvm/files/etc` to `podvm-mkosi` 
- Include Makefile.inc in the podvm-mkosi/Makefile to build the binaries (will need some adjustments)
- Move the fedora dockerfiles under podvm-mkosi as fedora is only used for podvm-mkosi builds

I plan to handle these in a separate PR as part of 1615. However, since this PR is touching the systemd files , can these files be copied to podvm-mkosi and used from there?

If it's too much work, then leave it. Will handle it separately.

Reply:
> Move the fedora dockerfiles under podvm-mkosi as fedora is only used for podvm-mkosi builds

okay for me, as you said this is independent and has no impact either way

> Copy the systemd config files in podvm/files/etc to podvm-mkosi
> Include Makefile.inc in the podvm-mkosi/Makefile to build the binaries (will need some adjustments)

I think duplicating this stuff will lead to a mess. There are no mkosi specific modifications to the Makefile.inc, so why would we duplicate that? Regarding the /files, I don't see benifeits either. The changes that this PR is doing must be made to the legacy podvm files, too, otherwise the legacy images will just break. There are no mkosi specific updates to these files. If you really don't want these changes in the legacy images, you would basically need need to fork guest-components and cloud-api-adaptor at least. 

In general, I don't think it is a good approach to isolate the two image-builds more than necessary from each other. We want to migrat -- making both approaches fully independent will rather lead to a fork than to a migration.

Reply:
I‘ll be able to take a more thorough look into it tomorrow, but I agree, this PR shouldn‘t break packer builds. (If that would be the case, we’d need more reviewers to ack) I‘m optimistic we‘re able introduce this iteratively without having to mirror everything.

Reply:
> Currently, I can't experiment with AI use cases involving large models and image sizes as it gets broken with this PR due to removal of the bind mount.

Ah, now this is actually a concrete problem we can work on. I [asked about the bind mounts some while ago in the slack ](https://cloud-native.slack.com/archives/C04A2EJ70BX/p1698158814429609), and my understanding after that conversation was that the mounts weren't used, that's why I removed them. There are some easy ways to unblock you in this regard:

- We can copy the files like you've said (fine for me if we have a reason like this)
- We can disable the units in the mkosi build
- We can remove the files in the docker build step of the binaries so they are not included in the tar ball (my preferred way)

> In parallel it gives us time to test F37 nvidia packages on F38 mkosi images and if it works, we enable it for mkosi built image. Or wait for F38 packages to be released to enable it in mkosi built image.

I would need to check, but I don't think the image has any f38 specific dependencies. We could try to downgrade the full image to f37, but notice that f37 is EOL since yesterday (if I understand [their schedule](https://fedorapeople.org/groups/schedule/f-39/f-39-key-tasks.html) correctly).

And we could also reconsider building both fedora + ubuntu images with mkosi if we can't get AI stuff running on fedora.

Reply:
> We can remove the files in the docker build step of the binaries so they are not included in the tar ball (my preferred way)

Since we clone + build the repo in the docker builds, I guess binaries + config files are somewhat coupled. so there might be inconsistencies if we retrieve /etc/**  from somewhere in the file tree (HEAD), while the binaries are copied from a docker layer (built from some specific ref).<

Since there is a plan to keep packer-based builds [in-tree](https://github.com/confidential-containers/cloud-api-adaptor/issues/1615#issuecomment-1843078257), for the time being, masking the units could be viable option.

Reply:
I've re-added the units in `/podvm` and added some lines to remove them during the mkosi build.